src/MBX/pair_mbx.cpp

```diff
@@ -0,0 +1,772 @@
+
+    // // for debugging
+
+    // pvector[8] = mbx_e2b_local;
+    // pvector[9] = mbx_e2b_ghost;
+    // pvector[10] = mbx_e3b_local;
+    // pvector[11] = mbx_e3b_ghost;
+    // pvector[12] = mbx_e4b_local;
+    // pvector[13] = mbx_e4b_ghost;
+    // pvector[14] = mbx_disp_real;
+    // pvector[15] = mbx_disp_pme;
+
+    // // for comparison with MBX
+
+    // pvector[16] = mbx_virial[0];
+    // pvector[17] = mbx_virial[1];
+    // pvector[18] = mbx_virial[2];
+    // pvector[19] = mbx_virial[3];
+    // pvector[20] = mbx_virial[4];
+    // pvector[21] = mbx_virial[5];
+  }
+}
+
+/* ----------------------------------------------------------------------
+   allocate all arrays
+------------------------------------------------------------------------- */
+
+void PairMBX::allocate()
+{
+  allocated = 1;
+  int n = atom->ntypes;
+
+  memory->create(setflag, n + 1, n + 1, "pair:setflag");
+  for (int i = 1; i <= n; i++)
+    for (int j = i; j <= n; j++) setflag[i][j] = 0;
+
+  memory->create(cutsq, n + 1, n + 1, "pair:cutsq");
+
+  memory->create(cut, n + 1, n + 1, "pair:cut");
+}
+
+/* ----------------------------------------------------------------------
+   global settings
+------------------------------------------------------------------------- */
+
+void PairMBX::settings(int narg, char **arg)
+{
+
+  //TODO Re-enable this once MBX is properly versioned at >1.3.2
+  // utils::logmesg(lmp, std::string("MBX Version ") + MBX_VERSION + "\n");
+
+  if (narg != 1) error->all(FLERR, "Illegal pair_style command");
+
+  cut_global = utils::numeric(FLERR, arg[0], false, lmp);
+
+  // reset cutoffs that have been explicitly set
+
+  if (allocated) {
+    int i, j;
+    for (i = 1; i <= atom->ntypes; i++)
+      for (j = i; j <= atom->ntypes; j++)
+        if (setflag[i][j]) cut[i][j] = cut_global;
+  }
+}
+
+/* ----------------------------------------------------------------------
+   set coeffs for one or more type pairs
+------------------------------------------------------------------------- */
+
+void PairMBX::coeff(int narg, char **arg)
+{
+  if (narg < 2) error->all(FLERR, "Incorrect num args for pair coefficients");
+  if (!allocated) allocate();
+
+  int ilo, ihi, jlo, jhi;
+  utils::bounds(FLERR, arg[0], 1, atom->ntypes, ilo, ihi, error);
+  utils::bounds(FLERR, arg[1], 1, atom->ntypes, jlo, jhi, error);
+
+  double cut_one = cut_global;
+  // if (narg == 5) cut_one = utils::numeric(FLERR, arg[4], false, lmp);
+
+  int count = 0;
+  for (int i = ilo; i <= ihi; i++) {
+    for (int j = MAX(jlo, i); j <= jhi; j++) {
+      cut[i][j] = cut_one;
+      setflag[i][j] = 1;
+      count++;
+    }
+  }
+
+  if (count == 0) error->all(FLERR, "Incorrect args for pair coefficients");
+
+  std::string fix_args = "";
+  for (int i = 2; i < narg; ++i) { fix_args += std::string(arg[i]) + " "; }
+
+  fix_args = fmt::format("_FIX_MBX_INTERNAL all MBX {}", fix_args);
+
+  fix_MBX = dynamic_cast<FixMBX *>(modify->add_fix(fix_args));
```

Comment:
Why create the fix in the coeff method? What happens if the user calls the pair_coeff command twice?

Reply:
I see. I would think passing the args for the fix as settings and then initializing the fix there would make more sense. Some example files have lots of `pair_coeff` calls which seems inefficient (and possibly confusing if the fix args differ between some `pair_coeff` calls).

Reply:
Ah I missed that. I was looking at in.mbx_mof_1h2o, but I see now that those calls are all for lj/cut or buck.